mlflow/utils/requirements_utils.py

```diff
@@ -409,3 +414,36 @@ def _get_pinned_requirement(package, version=None, module=None):
             version = version_raw
 
     return f"{package}=={version}"
+
+
+def _convert_package_name_to_module_name(package_name):
+    _init_modules_to_packages_map()
+
+    package_name = _normalize_package_name(package_name)
+
+    for module, pkg_list in _MODULES_TO_PACKAGES.items():
+        if package_name in pkg_list:
+            return module
```

Comment:
Is there a way to make this more efficient by constructing a reverse mapping from package name to module name up front? iterating through every entry in `importlib_metadata.packages_distributions()` every time this method is called may get expensive.